### ISSUE TITLE
infinite recursion in pcap_file.cpp

### ISSUE THREAD

**Issue by zarghul** [p4lang/behavioral-model#132]
There is a bug in the implementation of `src/bm_sim/pcap_file.cpp`.

In particular there is an infinite recursion chain of these functions:

`PcapFilesReader::scan() -> PcapFilesReader::schedulePacket() -> PcapFilesReader::timerFired() -> PcapFilesReader::scan()`

The recursion ends only when the pcap file reaches EOF.

A simple solution would be to add a loop in `scan()` and remove the recursive call to it in `timerFired()`:

```
--- pcap_file.cpp	2016-05-20 18:55:33.376173690 +0200
+++ new.cpp	2016-05-20 18:56:19.229550733 +0200
@@ -184,6 +184,7 @@
 
 void
 PcapFilesReader::scan() {
+  while (true) {
   if (nonEmptyFiles == 0) {
     BMLOG_DEBUG("Pcap reader: end of all input files");
     // notifyAllEnd();
@@ -237,6 +238,7 @@
 
   schedulePacket((unsigned)earliest_index, &delay);
 }
+}
 
 
 void PcapFilesReader::timerFired() {
@@ -251,7 +253,6 @@
   bool success = file->moveNext();
   if (!success)
     nonEmptyFiles--;
-  scan();
 }

```

**Comment by antoninbas** [comment]
Hi Yuri,

Thanks for reporting this. If you or your organization have signed the p4lang Apache CLA, feel free to submit a Pull Request here on Github (also send me an email with your full name and your organization name so I can check th CLA). If you haven't signed the CLA or don't want to submit a Pull Request, I can apply your patch myself.

**Comment by zarghul** [comment]
I just signed the CLA and sent you my info.

I also submitted a Pull Request: https://github.com/p4lang/behavioral-model/pull/133

**Comment by antoninbas** [comment]
fixed by #133
